### ISSUE TITLE
Support for HTTP footers

### ISSUE THREAD

**Issue by kali** [sozu-proxy/sozu#230]
Footers are THE killer feature of http. I don't understand why everybody hates them.

**Comment by divarvel** [comment]
Just to make sure, you mean support for Chunked encoding trailers, right?
(section 4.1.2 in RFC7230)

On Thu, Aug 24, 2017 at 6:04 PM, Mathieu Poumeyrol <notifications@github.com

**Comment by kali** [comment]
yes ! that and 4.4 ! i just love how it makes any attempt at a reasonable implementation of http fail.

**Comment by Geal** [comment]
my favorite is [pipelining](https://tools.ietf.org/html/rfc7230#section-6.3.2), which is only useful when someone wants to benchmark their brand new HTTP server. Nobody uses it in the real world, but everybody uses it in benchmarks, so they need to implement it.

**Comment by FlorentinDUBOIS** [comment]
Trailers are handled end-to-end on `main`: HTTP/2 trailers are decoded, validated and forwarded (`mux/pkawa.rs::handle_trailer`), H2→H1 conversion forwards them when the backend response is chunked (and elides them otherwise, per RFC 9110 §6.5), and HTTP/1 chunked trailers flow through kawa's `Trailers` parsing phase.

**Comment by kali** [comment]
👋
